sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -201,6 +202,7 @@ Connection getConnection() throws Exception {
   public abstract static class Read<T> extends PTransform<PBegin, PCollection<T>> {
     @Nullable abstract DataSourceConfiguration getDataSourceConfiguration();
     @Nullable abstract String getQuery();
+    @Nullable abstract List<Object> getParameters();
```

Comment:
I would stay with regular `java.sql.Type` even if it's a bit limited on some database. Maybe we can use the same approach as in the `Write`: a `StatementPreparator` interface where the user defines the parameters for the `PreparedStatement`. It would give a full control to the user.